lib/compat/wordpress-6.1/rest-api.php

```diff
@@ -93,3 +93,84 @@ function gutenberg_register_has_archive_on_post_types_endpoint() {
 	);
 }
 add_action( 'rest_api_init', 'gutenberg_register_has_archive_on_post_types_endpoint' );
+
+/**
+ * Use custom REST API Controller for Navigation Posts
+ *
+ * @param array $args the post type arguments.
+ * @return array the filtered post type arguments with the new REST Controller set.
+ */
+function gutenberg_update_navigation_rest_controller( $args, $post_type ) {
+	if ( in_array( $post_type, array( 'wp_navigation' ), true ) ) {
+		// Original set in
+		// https://github.com/WordPress/wordpress-develop/blob/6cbed78c94b9d8c6a9b4c8b472b88ee0cd56528c/src/wp-includes/post.php#L528.
+		$args['rest_controller_class'] = 'Gutenberg_REST_Navigation_Controller';
+	}
+	return $args;
+}
+add_filter( 'register_post_type_args', 'gutenberg_update_navigation_rest_controller', 10, 2 );
+
+function gutenberg_transform_slug_to_post_id( $response, $handler, WP_REST_Request $request ) {
+
+	// Create (POST) requests send the `slug` as part of the request **body** for the Post to be created.
+	// As a result the `slug` in POST requests should not be transformed into an ID.
+	$should_preserve_slug = 'POST' === $request->get_method();
+
+	// Ignore non-Navigation REST API requests.
+	if( ! str_starts_with( $request->get_route(), '/wp/v2/navigation' ) || $should_preserve_slug )  {
```

Comment:
We should check using `rest_get_route_for_post_type_items` instead of hardcoding `/wp/v2/navigation`.

Reply:
Thank you. Fixed ✅